code/assignTime.m

```diff
@@ -64,12 +64,28 @@
 diffIndices = find(packetGenTime_diffs < 0 );
 
 % Need to remove [diffIndices + 1], but may also need to remove subsequent
-% packets. Remove at most 5 adjacent packets (to prevent large un-needed
+% packets. Automatically remove the next packet [diffIndices + 2], as this is easier than
+% trying to confirm there is enough time to assign to samples without
+% causing overlap.
+% Remove at most 6 adjacent packets (to prevent large un-needed
 % packet rejection driven by positive outliers)
+numPackets = size(dataTable_original,1);
 indices_backInTime = [];
 for iIndex = 1:length(diffIndices)
-    counter = 1;
-    while (counter < 6) && dataTable_original.PacketGenTime(diffIndices(iIndex) + counter)...
+    counter = 3;
```

Comment:
may be add to the right of this line why counter = 3. I guess it has to do with the [diffIndices + 2]

Reply:
Resolved with 83db53bba4ad68982cbfeac256da13d0034a9baa